src/overlays/actors/ovl_En_Zf/z_en_zf.c

```diff
@@ -82,115 +151,2291 @@ static ColliderQuadInit D_80B4A204 = {
+                           !EnZf_PrimaryFloorCheck(this, globalCtx, 191.9956f) &&
+                           ((globalCtx->gameplayFrames % 2) == 0)) {
+                    EnZf_SetupJumpForward(this);
+                } else {
+                    EnZf_SetupApproachPlayer(this, globalCtx);
+                }
+            }
+        } else {
+            this->unk_3F0--;
+        }
+        if (curKeyFrame != (s32)this->skelAnime.curFrame) {
+            s32 nextKeyFrame = playSpeed + curKeyFrame;
+            if (((prevKeyFrame < 14) && (nextKeyFrame >= 16)) || ((prevKeyFrame < 27) && (nextKeyFrame >= 29))) {
+                Audio_PlayActorSound2(&this->actor, NA_SE_EN_RIZA_WALK);
+            }
+        }
+        if ((globalCtx->gameplayFrames & 0x5F) == 0) {
+            Audio_PlayActorSound2(&this->actor, NA_SE_EN_RIZA_CRY);
+        }
+    }
+}
+
+void EnZf_SetupDie(EnZf* this) {
+    Animation_Change(&this->skelAnime, &gZfDyingAnim, 1.5f, 0.0f, Animation_GetLastFrame(&gZfDyingAnim), 2, -4.0f);
+
+    if ((this->actor.bgCheckFlags & 1) && ((this->actor.velocity.y == 0.0f) || (this->actor.velocity.y == -4.0f))) {
+        this->actor.speedXZ = 0.0f;
+        this->hopAnimIndex = 0;
+    } else {
+        this->hopAnimIndex = 1;
+    }
+
+    this->action = ENZF_ACTION_DIE;
+    this->actor.flags &= ~1;
+
+    if (D_80B4A1B4 != -1) {
+        if (this->actor.prev != NULL) {
+            ((EnZf*)this->actor.prev)->unk_3F4 = 90;
+
+            if (this->actor.prev->colChkInfo.health < 3) {
+                this->actor.prev->colChkInfo.health = 3;
+            }
+        } else {
+            ((EnZf*)this->actor.next)->unk_3F4 = 90;
+
+            if (this->actor.next->colChkInfo.health < 3) {
+                this->actor.next->colChkInfo.health = 3;
+            }
+        }
+    }
+
+    D_80B4A1B0 = 0;
+    Audio_PlayActorSound2(&this->actor, NA_SE_EN_RIZA_DEAD);
+    EnZf_SetupAction(this, EnZf_Die);
+}
+
+void EnZf_Die(EnZf* this, GlobalContext* globalCtx) {
+
+    if (this->actor.bgCheckFlags & 2) {
+        this->actor.speedXZ = 0.0f;
+    }
+
+    if (this->actor.bgCheckFlags & 1) {
+        Math_SmoothStepToF(&this->actor.speedXZ, 0.0f, 1.0f, 0.15f, 0.0f);
+        this->hopAnimIndex = 0;
+    }
+
+    if (SkelAnime_Update(&this->skelAnime)) {
+        if (this->actor.category != ACTORCAT_PROP) {
+            if ((this->actor.params >= ENZF_TYPE_LIZALFOS_MINIBOSS_A) && (D_80B4A1B4 == -1)) {
+                Flags_SetSwitch(globalCtx, this->clearFlag);
+                func_800F5B58();
+            } else {
+                D_80B4A1B4 = -1;
+            }
+            Actor_ChangeCategory(globalCtx, &globalCtx->actorCtx, &this->actor, ACTORCAT_PROP);
+        }
+
+        if (this->alpha != 0) {
+            this->actor.shape.shadowAlpha = this->alpha -= 5;
+
+        } else {
+            Actor_Kill(&this->actor);
+            return;
```

Comment:
```suggestion
```

Reply:
This is normal in the codebase after an `Actor_Kill`, isn't it? There's one where I didn't put one, but it's right at the bottom of the function.

Reply:
it's normal when the alternative is using an else, but here it's equivalent to have or not have the return;, I think in these cases it's better not to have it

Reply:
yea to elaborate an early return makes sense in cases like
```
if (condition) {
    Actor_Kill
    return;
}
// Rest of the code
```
vs
```
if (condition) {
    Actor_Kill
} else {
    // Rest of the code
}
```
but here the return isnt doing anything, the kill is already in an else